### PR TITLE
Improve location info documentation

### PR DESCRIPTION
**Pull request by stevenvachon**
As discussed https://github.com/inikulin/parse5/issues/197#issuecomment-304474031

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -54,9 +54,9 @@ declare namespace MarkupData {
 declare namespace Options {
     export interface ParserOptions {
         /**
-         * Enables source code location information for the nodes. When enabled, each node (except root node) has the `__location` property.
-         * In case the node is not an empty element, `__location` will be {@link MarkupData.ElementLocation} object, otherwise it's {@link MarkupData.Location}.
-         * If the element was implicitly created by the parser it's `__location` property will be null.
+         * Enables source code location information. When enabled, each node (except the root node) will have a `__location` property.
+         * If the node is not an empty element, `__location` will be a {@link MarkupData.ElementLocation} object, otherwise it will be {@link MarkupData.Location}.
+         * If the element was implicitly created by the parser (as part of the [Adoption Agency Algorithm](https://html.spec.whatwg.org/multipage/syntax.html#adoption-agency-algorithm)), it's `__location` property will be `undefined`.
```
Comment (inikulin) [inline lib/index.d.ts]: The adoption agency algorithm is only one of many cases (though the trickiest one) when the parser creates elements implictly.

Comment (stevenvachon) [inline lib/index.d.ts]: If it's not too much to ask, could you list them all here? I'll try to summarize them tersely.

Comment (inikulin) [inline lib/index.d.ts]: It's a lot of cases. I believe we shouldn't list them all in documentation either. Just give an example (https://html.spec.whatwg.org/multipage/syntax.html#an-introduction-to-error-handling-and-strange-cases-in-the-parser has a good explanation for some such cases, you can point to it).

Comment (stevenvachon) [inline lib/index.d.ts]: Done.

**Comment by inikulin** [comment]
Thank you!

**Pull request merged**